lib/connection.ts

```diff
@@ -349,23 +386,33 @@ export class Connection {
   }
 
   /**
-   * Registers the given handler for the specified event.
-   * @param {rhea.ConnectionEvents} event The event for which the handler needs to be registered.
-   * @param {rhea.OnAmqpEvent} handler The handler function that needs to be executed when the event
-   * occurs.
-   * @returns {void} void.
+   * Adds event listeners for the possible events that can occur on the connection object and
+   * re-emits the same event back with the received arguments from rhea's event emitter.
+   * @private
+   * @returns {void} void
    */
-  registerHandler(event: ConnectionEvents, handler: rhea.OnAmqpEvent): void {
-    this._connection.on(event, handler);
-  }
+  private _initializeEventListeners(): void {
+    for (const eventName in ConnectionEvents) {
+      this._connection.on(ConnectionEvents[eventName],
+        (...args) => this.emit(ConnectionEvents[eventName], ...args));
+    }
 
-  /**
-   * Removes the given handler for the specified event.
-   * @param {rhea.ConnectionEvents} event The event for which the handler needs to be removed.
-   * @param {rhea.OnAmqpEvent} handler The handler function that needs to be removed.
-   * @returns {void} void.
-   */
-  removeHandler(event: ConnectionEvents, handler: rhea.OnAmqpEvent): void {
-    this._connection.removeListener(event, handler);
+    // Add event handlers for *_error and *_close events that can be propogated to the connection
```

Comment:
didn't we decide that events won't end up propagating up because they'll automatically be listened to at the lower level? i would expect to see listeners get set up for the connection events specifically rather than link/session events

Reply:
They won't. However, I added them for additional safety, just in case.